### PR TITLE
fix(s3-url-builder): sanitize attachment name

### PR DESCRIPTION
**Pull request by sagebind**
DAM-8293

### CODE REVIEW

File: src/com/widen/util/S3UrlBuilder.java [hunk 1]
```diff
@@ -330,7 +332,11 @@ else if (BucketEncoding.VIRTUAL_DNS.equals(requestedBucketEncoding))
 		{
 			canSign();
 
-			builder.addParameter("response-content-disposition", String.format("attachment; filename=\"%s\"", attachmentFilename));
+			// Most browsers don't support UTF-8 in HTTP headers (HTTP officially supports only ISO-8859-1). In
+			// practice, S3 only supports ASCII characters, so strip everything out not in the ASCII range.
+			String strippedFilename = attachmentFilename.replaceAll("[^\\x00-\\x7F]", "");
```
Comment (uriahcarpenter) [inline src/com/widen/util/S3UrlBuilder.java]: Running the name though Normalizer would help preserve some characters... http://stackoverflow.com/a/17786019/96224

Regex could be tightened to strip control characters: `[^\\x20-\\x7E]`

**Comment by uriahcarpenter** [comment]
Attachment names also at:

 - https://github.com/Widen/urlbuilder/blob/master/src/com/widen/util/CloudfrontUrlBuilder.java#L121

File: src/com/widen/util/S3UrlBuilder.java (hunk 1 above)
Comment (uriahcarpenter) [inline src/com/widen/util/S3UrlBuilder.java]: Also please add some tests to exercise this code.

Comment (sagebind) [inline src/com/widen/util/S3UrlBuilder.java]: Normalizing is a good plan, I'll update the PR and also add some tests.

**Comment by uriahcarpenter** [comment]
Looking good. I'd add another InternalUtils test of the accent chars conversion: e.g. `é` --> `e`

I think we can make this release be 1.0.0 -- and cleanup the readme pre-release notes.

**Comment by uriahcarpenter** [comment]
One corner case is what happens if *all* text is out of the ASCII range? Is an empty string best, so should some random (or hash) text be used instead?

**Comment by sagebind** [comment]
I would say to not specify a filename if the cleaned filename is blank, so the filename used would just be determined by S3.

**Comment by uriahcarpenter** [comment]
Looks good to me. Final change would be to move [this](https://github.com/Widen/urlbuilder/pull/2/files#diff-0cca7efbc63f6f060fbb0eceaa65ad09R335) comment to `InternalUtils `
